src/lib/fetch.ts

```diff
@@ -1,100 +1,385 @@
 import 'isomorphic-fetch';
+import { Field } from '../snarky';
+import { UInt32, UInt64 } from './int';
+import { Permission, Permissions, ZkappStateLength } from './party';
+import { PublicKey } from './signature';
 
-type Config = {
-  timeout: number | undefined;
+export {
+  fetchAccount,
+  parseFetchedAccount,
+  markAccountToBeFetched,
+  fetchMissingAccounts,
+  getCachedAccount,
+  addCachedAccount,
+  defaultGraphqlEndpoint,
+  setGraphqlEndpoint,
+  sendZkappQuery,
+  sendZkapp,
 };
+export { Account };
 
-type FetchResponse = { data: any };
+let defaultGraphqlEndpoint = 'none';
+function setGraphqlEndpoint(graphqlEndpoint: string) {
+  defaultGraphqlEndpoint = graphqlEndpoint;
+}
+
+/**
+ * Gets account information on the specified publicKey by performing a GraphQL query
+ * to the specified endpoint. This will call the 'GetAccountInfo' query which fetches
+ * zkapp related account information.
+ *
+ * If an error is returned by the specified endpoint, an error is thrown. Otherwise,
+ * the data is returned.
+ *
+ * @param publicKey The specified account to get account information on
+ * @param graphqlEndpoint The graphql endpoint to fetch from
+ * @param config An object that exposes an additional timeout option
+ * @returns zkapp information on the specified account or an error is thrown
+ */
+async function fetchAccount(
```

Comment:
should we also be checking the account cache here in addition to writing to it?

Reply:
hmm, I feel that it's nice to have one explicit method to definitely perform the fetch and bypass the cache

Reply:
Fair